shark_turbine/kernel/wave/promotion.py

```diff
@@ -0,0 +1,41 @@
+from ...support.logging import get_logger
+from shark_turbine.kernel._support.tracing import CapturedTrace
+import torch.fx as fx
+from ..ops.wave_ops import *
+from .address_spaces import *
+import shark_turbine.kernel.lang as tkl
+
+logger = get_logger("turbine.wave.promotion")
+
+
+def apply_promotion_pattern_(custom_node: Read | Write, allocate_node: Allocate):
```

Comment:
nit: Why does the name have a trailing _?

Reply:
Same as above, will change.